e2e/url/url.go

```diff
@@ -18,6 +18,20 @@ type URL struct {
 	port uint16
 }
 
+func New(address string) (*URL, error) {
```

Comment:
nit: can we rename this to `ParseString`?

Reply:
:+1: b74b44d